addons/sale_purchase/models/sale_order.py

```diff
@@ -17,12 +17,25 @@ def _compute_purchase_order_count(self):
         for order in self:
             order.purchase_order_count = len(order._get_purchase_orders())
 
+    def _set_analytic_account(self):
+        purchase_orders = self._get_purchase_orders()
+        purchase_order_lines = purchase_orders.order_line.filtered(lambda line: line.sale_order_id == self)
+        for order_line in purchase_order_lines:
+            order_line.analytic_distribution = {self.analytic_account_id.id: 100}
+
     def _action_confirm(self):
         result = super(SaleOrder, self)._action_confirm()
         for order in self:
             order.order_line.sudo()._purchase_service_generation()
+            if order.analytic_account_id:
+                self._set_analytic_account()
```

Comment:
I don't think this check is needed since the order in self will be given an ``analytic_account_id`` from the super call.